### PR TITLE
Support includes and matches

### PR DESCRIPTION
**Pull request by odin568**
Harmonized example config to match readme
Updated Readme

References to  #19

### CODE REVIEW

**Comment by odin568** [comment]
Implemented without being really able to run locally - only inside docker. That's why it lacks the test for now.

File: lib/ical_filter_proxy/filter_rule.rb [hunk 1]
```diff
@@ -27,6 +27,10 @@ def evaluate(event_data, value)
           event_data == value
         when 'startswith'
           event_data.start_with?(value)
+        when 'includes'
+          event_data.include?(value)
+        when 'matches'
+          event_data =~ eval(value)
```
Comment (darkphnx) [inline lib/ical_filter_proxy/filter_rule.rb:33]: `eval` is a pretty dangerous method to be using here. Instead use `Regexp.new` to compile your regular expression.

**Review by darkphnx — CHANGES_REQUESTED** [review]
Looks promising. If you put together what you think is a correct test and push, it will be run through CI. Not ideal, but easier than installing ruby etc. if you don't have that already.

Alternatively, feel free to put together a development docker image after I've fixed the main one.

File: lib/ical_filter_proxy/filter_rule.rb (hunk 1 above)
Comment (odin568) [reply lib/ical_filter_proxy/filter_rule.rb:33]: Makes sense, will do and update

Comment (odin568) [reply lib/ical_filter_proxy/filter_rule.rb]: The only good thing about eval was that I could add /<pattern>/<options>.
With Regex.new I have do pass <pattern> and the options might be "hard coded".
You see any chance or have any preference?

Comment (odin568) [reply lib/ical_filter_proxy/filter_rule.rb]: Updated with a proposal, please have a look @darkphnx

Comment (darkphnx) [reply lib/ical_filter_proxy/filter_rule.rb]: It's not very pretty, but you could parse a // expression yourself and set options in Regexp.new. Something like:

```ruby
REGEXP_OPTS_LOOKUP = { 
  'i' => Regexp::IGNORECASE, 
  'm' => Regexp::MULTILINE, 
  'x' => Regexp::EXTENDED,
}

def match_value_to_regexp(value) 
  parsed_value = /^\/(?<pattern>.+)\/(?<opts>.*)$/.match(value)

  opts = parsed_value[:opts]
    .chars
    .map { |opt| REGEXP_OPTS_LOOKUP[opt] }
    .compact
    .inject { |memo, opt| memo | opt }

  Regexp.new(parsed_value[:pattern], opts)
end

value = "/my-search/i"
regexp = match_value_to_regexp(value)
# => /my-search/i
```

**Review by darkphnx — COMMENTED** [review]
That's an acceptable workaround. Pop a couple of tests in and I'll get it merged.

**Comment by odin568** [comment]
Did some testing. Can you have a look if that is sufficient?

**Review by darkphnx — APPROVED** [review]
Beautiful!

**Pull request merged**